sqlmesh/dbt/adapter.py

```diff
@@ -309,13 +317,38 @@ def resolve_identifier(self, relation: BaseRelation) -> t.Optional[str]:
         return identifier
 
     def _map_table_name(
-        self, database: t.Optional[str], schema: t.Optional[str], identifier: t.Optional[str]
+        self,
+        database: t.Optional[str | exp.Identifier],
+        schema: t.Optional[str | exp.Identifier],
+        identifier: t.Optional[str | exp.Identifier],
     ) -> exp.Table:
-        name = ".".join(p for p in (database, schema, identifier) if p is not None)
+        name = ".".join(
+            p.name if isinstance(p, exp.Identifier) else p
+            for p in (database, schema, identifier)
+            if p is not None
+        )
         if name not in self.table_mapping:
             return exp.to_table(name, dialect=self.engine_adapter.dialect)
 
         physical_table_name = self.table_mapping[name]
         logger.debug("Resolved ref '%s' to snapshot table '%s'", name, physical_table_name)
 
         return exp.to_table(physical_table_name, dialect=self.engine_adapter.dialect)
+
+    @t.overload
+    def _normalize(self, name: None, quoted: bool = True) -> None:
+        ...
+
+    @t.overload
+    def _normalize(self, name: str, quoted: bool = True) -> exp.Identifier:
+        ...
+
+    def _normalize(self, name: t.Optional[str], quoted: bool = True) -> t.Optional[exp.Identifier]:
```

Comment:
why doesn't normalize take in like a table or something? that way you don't have to call it multiple times, or it can take multiple args

Reply:
done